### PR TITLE
Fabo/2683 refactor blocks throttle

### LINKED ISSUE #2683: Direct navigating or refreshing on or beyond /staking/validators does not load data
**Describe the bug**
When at https://lunie.io/#/staking/validators/ or any sub url refreshes do not fetch data. This also affects direct links. Such as a validator trying to link to: https://lunie.io/#/staking/validators/cosmosvaloper1sjllsnramtg3ewxqwwrwjxfgc4n4ef9u2lcnj0 but being unable to.

**To Reproduce**
1. Click "Staking" from the side navbar
2. Refresh the page and no validator info will load.
2.1 Also direct links to validators are affected by presumably the same issue.

**Expected behavior**
Should refresh and see content or be able to link directly to the validator.

**Screenshots**
Refresh from route: /staking/validators/
Screen Shot 2019-06-05 at 4 08 50 PM

Direct link to any validator: 
Screen Shot 2019-06-05 at 4 09 16 PM

**Desktop (please complete the following information):**
 - OS: 
 - Browser: 
 - Version: 

**Smartphone (please complete the following information):**
 - Device: 
 - OS: 
 - Browser 
 - Version 

**Additional context**

### PR DESCRIPTION
**Pull request by faboweb**
Closes #2683

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by jbibla** [comment]
i don't understand what's happening here. can you explain?

File: src/components/staking/TabMyDelegations.vue [hunk 1]
```diff
@@ -137,7 +137,6 @@ export default {
     }
   },
   async mounted() {
-    this.$store.dispatch(`updateDelegates`)
```
Comment (faboweb) [inline src/components/staking/TabMyDelegations.vue]: Only the TableValidator is requesting validators

File: src/components/staking/TabValidators.vue [hunk 2]
```diff
@@ -38,9 +38,6 @@ export default {
     ])
   },
   watch: {
-    "session.signedIn": function(signedIn) {
-      signedIn && this.$store.dispatch(`updateDelegates`)
```
Comment (faboweb) [inline src/components/staking/TabValidators.vue]: Only the TableValidator is requesting validators

**Comment by faboweb** [comment]
I added comments. This is also just a refactor and cleanup the existing (intended) behavior doesn't change. We prevent requerying of data for x blocks for a couple of datasets.

**Comment by colw** [comment]
Wwhat drives the delegates to update? There are still 3 other locations where it is called? Could it be reduce to one? Once the store is updated the components will take care of themselves.

**Comment by colw** [comment]
I mean rather than adding code to throttle one thing, we limit when it's called.

It feels like a kludge :)

**Comment by faboweb** [comment]
We could throttle this in the components. But then what happens if component A mounts and calls it, then component B mounts and calls it then component C mounts and calls it. Now we loaded the validators 3 times. With the centralized approach each component just requests the validators. The store checks if it has fresh or stale data. How would you do it?

**Comment by faboweb** [comment]
Sorry my naming was really bad, I improved it.

**Comment by colw** [comment]
I mean to not have the lower components call it at all.

**Comment by faboweb** [comment]
I liked your arguments and we might restructure this soon.

**Comment by jbibla** [comment]
does this close #2252

**Pull request merged**